userspace/libsinsp/CMakeLists.txt

```diff
@@ -97,6 +90,16 @@ set(SINSP_SOURCES
 	viewinfo.cpp
 )
 
+if(WITH_CHISEL)
+	list(APPEND SINSP_SOURCES
+		../chisel/chisel_api.cpp
+		../chisel/chisel_fields_info.cpp
+		../chisel/chisel_utils.cpp
+		../chisel/chisel.cpp
+		../chisel/lua_parser_api.cpp
+		../chisel/lua_parser.cpp)
+endif()
```

Comment:
Could this be a separate `libsinspchisel.a`? We'd lose one build-time variable for libsinsp, at least

Reply:
*Short answer*
it could be, but it's outside the scope of this PR :)

*Long answer*
The main goal of this PR is to reduce the external dependencies by moving out those sub-components (which should not part of `libsinsp` by default). When doing that, I've realized there are a lot of further improvements that can be done, but the set of modifications required just to move them away was already too big. In particular, I have noticed they are tightly coupled with `libsinsp` internals, most of the implementation relies on friend classes, unfortunately. So I decided to keep things simple and reduce the scope of this PR as minimal as possible (that's XXL size anyways). Basically, you should think about this PR as a baby step.
For those reasons, I really think that we should first review the implementation and then make `libsinspchisel.a` a thing, eventually.